src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -113,8 +113,10 @@ export class ExamScoresComponent implements OnInit {
             const exerciseResult = studentResult.exerciseGroupIdToExerciseResult[exerciseGroup.id];
             if (exerciseResult) {
                 csvRow[exerciseGroup.title + 'AssignedExercise'] = exerciseResult.title ? exerciseResult.title : '';
-                csvRow[exerciseGroup.title + 'AchievedPoints'] = exerciseResult.achievedPoints ? this.round(exerciseResult.achievedPoints, 1) : '';
-                csvRow[exerciseGroup.title + 'AchievedScore(%)'] = exerciseResult.achievedScore ? this.round(exerciseResult.achievedScore, 2) : '';
+                csvRow[exerciseGroup.title + 'AchievedPoints'] =
+                    typeof exerciseResult.achievedPoints === 'undefined' || exerciseResult.achievedPoints === null ? '' : this.round(exerciseResult.achievedPoints, 1);
+                csvRow[exerciseGroup.title + 'AchievedScore(%)'] =
```

Comment:
You can also use just `exerciseResult.achievedPoints == null`. I think the `typeof` can cause problems with null as it returns an object.
See https://stackoverflow.com/questions/2647867/how-can-i-determine-if-a-variable-is-undefined-or-null